lib/CallBackery/qooxdoo/callbackery/source/class/callbackery/ui/plugin/ActionForm.js

```diff
@@ -0,0 +1,312 @@
+                            break;
+                        case 'cancel':
+                            this.fireDataEvent('actionResponse',{action: 'cancel'});
+                            break;
+                        case 'popup':
+                            if (! btCfg.noValidation) { // backward incompatibility work around
+                                var formData = getParentFormData();
+                                if (formData === false){
+                                    callbackery.ui.MsgBox.getInstance().error(
+                                        this.tr("Validation Error"),
+                                        this.tr("The form can only be submitted when all data fields have valid content.")
+                                    );
+                                    return;
+                                }
+                            }
+                            var popup = new callbackery.ui.Popup(btCfg,getParentFormData, this);
+                            var appRoot = that.getApplicationRoot();
+                            popup.addListenerOnce('close',function(){
+                                // wait for stuff to happen before we rush into
+                                // disposing the popup
+                                qx.event.Timer.once(function(){
+                                    appRoot.remove(popup);
+                                    popup.dispose();
+                                    this.fireEvent('popupClosed');
+                                },that,100);
+                            },that);
+                            popup.open();
+                            break;
+                        case 'logout':
+                            that.fireDataEvent('actionResponse',{action: 'logout'});
+                            break;
+
+                        default:
+                            this.debug('Invalid execute action:' + btCfg.action);
+                    }
+                }; // var action = function() { ... };
+
+                if (btCfg.defaultAction){
+                    this._defaultAction = action;
+                }
+                if (button){
+                    button.addListener('execute',action,this);
+                    if (btCfg.addToMenu) {
+                        menues[btCfg.addToMenu].add(button);
+                    }
+                    else {
+                        if (btCfg.addToToolBar !== false) {
+                            if (btCfg.action == 'cancel') {
+                                this._btnRow.add(button);
+                            }
+                            else {
+                                this._grid.setRowAlign(row, 'left', 'middle');
+                                this._form.add(button, {row : row, column : btCfg.btnColumn || colBtn});
+                                if (btCfg.description) {
+                                    description = new qx.ui.basic.Label(this.xtr(btCfg.description || ''));
```

Comment:
this only happens if there is a description ... so I guess the `|| ''` is not necessary

Reply:
Removed